### PR TITLE
identity provider UI

### PR DESCRIPTION
**Pull request by jgruica**
Screen Shot 2020-12-04 at 11 46 20 AM
Screen Shot 2020-12-04 at 11 57 55 AM
Screen Shot 2020-12-04 at 9 08 48 AM

### CODE REVIEW

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+        </div> */}
+        <form className="flex flex-column Identity--wrapper u-marginTop--30">
+          <p className="u-fontSize--largest u-lineHeight--default u-fontWeight--bold u-color--tuna"> Configure Identity Provider </p>
+          <p className="u-fontSize--normal u-lineHeight--medium u-fontWeight--medium u-color--dustyGray u-marginTop--12"> Configure additional ODIC providers to authenticate in to the Admin Console. </p>
+
+          <div className="u-marginTop--30">
+            <div className="flex flex1 alignItems--center">
+              <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> Admin Console URL </p>
+              <span className="required-label"> Required </span>
+              {requiredErrors?.adminConsoleAddress && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:286]: "Admin Console URL is a required field"

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+              placeholder="kots.somebigbankadmin.com"
+              value={this.state.adminConsoleAddress}
+              onChange={(e) => { this.handleFormChange("adminConsoleAddress", e) }} />
+          </div>
+
+          <div className="u-marginTop--30">
+            <div className="flex flex1 alignItems--center">
+              <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> ID Address </p>
+              <span className="required-label"> Required </span>
+              {requiredErrors?.identityServiceAddress && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:299]: Same here, "`fieldName` is a required field"

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+              </label>
+            </div>
+          </div>
+
+          {selectedProvider === "oidcConfig" &&
+            <div className="u-marginTop--30">
+              <div className="flex flex1 alignItems--center">
+                <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> Connector  name </p>
+                <span className="required-label"> Required </span>
+                {requiredErrors?.connectorName && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:337]: Same here, "`fieldName` is a required field"

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+                placeholder="connector name"
+                value={this.state.oidcConfig.connectorName}
+                onChange={(e) => { this.handleFormChange("connectorName", e) }} />
+            </div>}
+
+          <div className="u-marginTop--30">
+            <div className="flex flex1 alignItems--center">
+              <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> Issuer </p>
+              <span className="required-label"> Required </span>
+              {requiredErrors?.issuer && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:350]: Same here, "`fieldName` is a required field"

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+              placeholder="kots.somebigbankadmin.com"
+              value={selectedProvider === "oidcConfig" ? this.state.oidcConfig?.issuer : this.state.geoAxisConfig?.issuer ? this.state.geoAxisConfig?.issuer : ""}
+              onChange={(e) => { this.handleFormChange("issuer", e) }} />
+          </div>
+
+          <div className="u-marginTop--30">
+            <div className="flex flex1 alignItems--center">
+              <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> Client ID </p>
+              <span className="required-label"> Required </span>
+              {requiredErrors?.clientId && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:363]: Same here, "`fieldName` is a required field"

File: kotsadm/web/src/components/identity/IdentityProviders.jsx [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+              placeholder="client ID"
+              value={selectedProvider === "oidcConfig" ? this.state.oidcConfig?.clientId : this.state.geoAxisConfig?.clientId ? this.state.geoAxisConfig?.clientId : ""}
+              onChange={(e) => { this.handleFormChange("clientId", e) }} />
+          </div>
+
+          <div className="u-marginTop--30">
+            <div className="flex flex1 alignItems--center">
+              <p className="u-fontSize--large u-lineHeight--default u-fontWeight--bold u-color--tuna"> Client Secret </p>
+              <span className="required-label"> Required </span>
+              {requiredErrors?.clientSecret && <span className="u-color--chestnut u-fontSize--small u-fontWeight--medium u-lineHeight--normal u-marginLeft--5"> This item is required </span>}
```
Comment (GraysonNull) [inline kotsadm/web/src/components/identity/IdentityProviders.jsx:376]: Same here, "`fieldName` is a required field"

**Review by GraysonNull — CHANGES_REQUESTED** [review]

File: kotsadm/web/src/components/identity/AnnotationRow.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+              <input type="text"
+                className="Input darker"
+                style={{ marginLeft: "12px" }}
+                placeholder="value"
+                value={this.state.value}
+                onChange={(e) => { this.handleFormChange("value", e) }} />
+            </div>
+          </div>
+          <div className="flex fle1 justifyContent--flexEnd">
+            <span className="icon gray-trash u-cursor--pointer" onClick={this.props.removeAnnotation} />
```
[the remark this answers is not in the document]
Comment (jgruica) [reply kotsadm/web/src/components/identity/AnnotationRow.jsx:40]: this component is still WIP. we're not showing those rows now. (it's part of configuring ingress)

**Review by GraysonNull — APPROVED** [review]

**Pull request merged**